botorch/acquisition/bayesian_active_learning.py

```diff
@@ -0,0 +1,248 @@
+        [mackay1992alm]_.
+
+        Args:
+            model: A fully bayesian single-outcome model.
+            X_pending: A `batch_shape, m x d`-dim Tensor of `m` design points.
+        """
+        super().__init__(model)
+        self.set_X_pending(X_pending)
+
+    @concatenate_pending_points
+    @t_batch_mode_transform()
+    def forward(self, X: Tensor) -> Tensor:
+        posterior = self.model.posterior(X, observation_noise=True)
+        res = torch.logdet(posterior.mixture_covariance_matrix).exp()
+
+        # the MCMC dim is averaged out in the mixture postrior,
+        # so the result needs to be unsqueeze[d for the averaging
+        # in the decorator
+        return res.unsqueeze(-1)
+
+
+class qBayesianQueryByComittee(FullyBayesianAcquisitionFunction):
+    def __init__(
+        self,
+        model: SaasFullyBayesianSingleTaskGP,
+        X_pending: Optional[Tensor] = None,
+    ) -> None:
+        """
+        Bayesian Query-By-Comittee [riis2022fbgp]_, which minimizes the variance
+        of the mean in the posterior.
+
+        Args:
+            model: A fully bayesian single-outcome model.
+            X_pending: A `batch_shape, m x d`-dim Tensor of `m` design points
+        """
+        super().__init__(model)
+        self.set_X_pending(X_pending)
+
+    @concatenate_pending_points
+    @t_batch_mode_transform()
+    def forward(self, X: Tensor) -> Tensor:
+        posterior = self.model.posterior(X)
+        posterior_mean = posterior.mean
+        marg_mean = posterior.mixture_mean.unsqueeze(MCMC_DIM)
+        mean_diff = posterior_mean - marg_mean
+        covar_of_mean = torch.matmul(mean_diff, mean_diff.transpose(-1, -2))
+
+        res = torch.logdet(covar_of_mean).exp()
+        return torch.nan_to_num(res, 0)
+
+
+class qBayesianActiveLearningByDisagreement(
+    FullyBayesianAcquisitionFunction, MCSamplerMixin
+):
+    def __init__(
+        self,
+        model: SaasFullyBayesianSingleTaskGP,
+        X_pending: Optional[Tensor] = None,
+        estimation_type: str = "LB",
+        sampler: Optional[MCSampler] = None,
+        num_samples: int = 64,
+    ) -> None:
+        """Batch implementation [kirsch2019batchbald]_ of BALD [houlsby2011bald]_,
+        which maximizes the mutual information between the next observation and the
+        hyperparameters of the model.
+
+        Args:
+            model: A fully bayesian single-outcome model.
+            X_pending: A `batch_shape, m x d`-dim Tensor of `m` design points
+            estimation_type: estimation_type: A string to determine which entropy
+                estimate is computed: Lower bound" ("LB") or "Monte Carlo" ("MC")
+                (not implemented yet).
+            sampler: MCSampler for Monte Carlo estimation of the statistical distance
+                (not implemented yet).
+            num_samples (int, optional): Number of samples if employing monte carlo
+                estimation of the statistical distance. Defaults to 64.
```

Comment:
Let's remove the args for not implemented functionality and add them when they're implemented?